### PR TITLE
update jsonrpsee

### PR DESCRIPTION
**Pull request by CHr15F0x**
This PR breaks API versioning and will be followed by another PR that fixes it.

### CODE REVIEW

File: crates/pathfinder/src/bin/pathfinder/main.rs [hunk 1]
```diff
@@ -320,7 +320,7 @@ If you are trying to setup a custom StarkNet please use '--network custom',
                 Err(err) => tracing::error!("Cairo process ended unexpected; failed to join task handle: {:?}", err),
             }
         }
-        _result = rpc_handle => {
+        _result = rpc_handle.stopped() => {
```
Comment (Mirko-von-Leipzig) [inline crates/pathfinder/src/bin/pathfinder/main.rs:323]: Does this return a more meaningful error now that we can emit on crash?

Previously just returned `()` which is of course useless.

File: crates/rpc/src/lib.rs [hunk 2]
```diff
@@ -42,9 +42,9 @@ impl RpcServer {
     }
 
     /// Starts the HTTP-RPC server.
-    pub async fn run(self) -> Result<(HttpServerHandle, SocketAddr), anyhow::Error> {
-        let server = HttpServerBuilder::default()
-            .set_middleware(self.middleware)
+    pub async fn run(self) -> Result<(ServerHandle, SocketAddr), anyhow::Error> {
+        let server = ServerBuilder::default()
+            .set_logger(self.middleware)
```
Comment (Mirko-von-Leipzig) [inline crates/rpc/src/lib.rs:47]: Should we rename `self.middleware` to `self.logger`.

Comment (Mirko-von-Leipzig) [reply crates/rpc/src/lib.rs:47]: Or is it actually our `metrics` middleware?

File: crates/pathfinder/src/bin/pathfinder/main.rs (hunk 1 above)
Comment (CHr15F0x) [reply crates/pathfinder/src/bin/pathfinder/main.rs:323]: I'm afraid it's still the same case now, i.e. `()` .

File: crates/rpc/src/lib.rs (hunk 2 above)
Comment (CHr15F0x) [reply crates/rpc/src/lib.rs:47]: Looks like in the meantime `jsonrpsee` switched from _middleware_ to _logger_ to describe the functionality of adding callbacks that allow you to take notice of certain events. This happened probably due to the fact that at that time they needed to differentiate from _middleware_ understood as `tower` _middleware_ which `jsonrpsee` started supporting.

So, yes, it is our `metrics` middleware. I could rename it all to logger (`self.middleware` to `self.logger`, `metrics::middleware::RpcMetricsMiddleware` to `metrics::logger::RpcMetricsLogger`, and so on). WDYT?

Comment (Mirko-von-Leipzig) [reply crates/rpc/src/lib.rs:47]: Hmm honestly your call - whatever you think makes sense. I guess the type signature will help no matter what.

**Review by kkovaacs — APPROVED** [review]
LGTM

**Pull request merged**